utils/check_repo.py

```diff
@@ -905,19 +897,26 @@ def find_all_documented_objects() -> List[str]:
 
     Returns:
         `List[str]`: The list of all object names being documented.
+        `Dict[str, List[str]]`: A dictionary mapping the object name (full import path, e.g.
+            `integrations.PeftAdapterMixin`) to its documented methods
     """
     documented_obj = []
-    for doc_file in Path(PATH_TO_DOC).glob("**/*.rst"):
-        with open(doc_file, "r", encoding="utf-8", newline="\n") as f:
-            content = f.read()
-        raw_doc_objs = re.findall(r"(?:autoclass|autofunction):: transformers.(\S+)\s+", content)
-        documented_obj += [obj.split(".")[-1] for obj in raw_doc_objs]
+    documented_methods_map = {}
     for doc_file in Path(PATH_TO_DOC).glob("**/*.md"):
         with open(doc_file, "r", encoding="utf-8", newline="\n") as f:
             content = f.read()
         raw_doc_objs = re.findall(r"\[\[autodoc\]\]\s+(\S+)\s+", content)
         documented_obj += [obj.split(".")[-1] for obj in raw_doc_objs]
-    return documented_obj
+
+        for obj in raw_doc_objs:
+            obj_public_methods = re.findall(rf"\[\[autodoc\]\] {obj}((\n\s+-.*)+)", content)
+            # Some objects have no methods documented
+            if len(obj_public_methods) == 0:
+                continue
+            else:
+                documented_methods_map[obj] = re.findall(r"(?<=-\s).*", obj_public_methods[0][0])
```

Comment:
note: I intend to use this mapping in a follow-up PR, where we check the quality of the docstrings in this mapping (i.e. the docstrings of all public methods, currently without quality checks)